tests/test_types.py

```diff
@@ -2002,3 +2002,90 @@ def test_dumping_passes_context_to_inner_type_dump(self):
         context = object()
         DumpOnly(inner_type).dump('foo', context)
         assert inner_type.dump_context == context
+
+
+class TestTransform:
+    def test_loading_calls_pre_load_with_original_value(self):
+        class Callbacks():
+            def pre_load(self, data):
+                self.pre_loaded = data
+                return data
+        callbacks = Callbacks()
+        Transform(SpyType(), pre_load=callbacks.pre_load).load('foo')
+        assert callbacks.pre_loaded == 'foo'
+
+    def test_dump_calls_pre_dump_with_original_value(self):
+        class Callbacks():
+            def pre_dump(self, data):
+                self.pre_dump = data
+                return data
+        callbacks = Callbacks()
+        Transform(SpyType(), pre_dump=callbacks.pre_dump).dump('foo')
+        assert callbacks.pre_dump == 'foo'
+
+    def test_loading_calls_inner_type_load_with_result_of_pre_load(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, pre_load=lambda foo: 'bar')
+        transformed = transform.load('foo')
+        assert inner_type.loaded == 'bar'
+
+    def test_loading_calls_post_load_with_result_of_inner_type_load(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, post_load=lambda foo: foo + 'bar')
+        transformed = transform.load('foo')
+        assert inner_type.loaded == 'foo'
+        assert transform.load('foo') == 'foobar'
```

Comment:
You do not need to call `load` again since you have already captured return value. How about this:
```python
result = transform.load('foo')
assert result == 'foobar'
assert inner_type.loaded == 'foo'
```